subsys/net/lib/sockets/sockets_inet.c

```diff
@@ -1061,6 +1061,68 @@ static int add_pktinfo(struct net_context *ctx,
 	return ret;
 }
 
+static int add_hoplimit(struct net_context *ctx,
+			struct net_pkt *pkt,
+			struct msghdr *msg)
+{
+	int ret = -ENOTSUP;
+	struct net_pkt_cursor backup;
+
+	net_pkt_cursor_backup(pkt, &backup);
+	net_pkt_cursor_init(pkt);
+
+	if (IS_ENABLED(CONFIG_NET_IPV4) && net_pkt_family(pkt) == AF_INET) {
+		NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE(ipv4_access,
```

Comment:
When processing IPv4 the stack sets the TTL value in the net_pkt, I think it would be way easier to read it with `net_pkt_ipv4_ttl()` instead fetching the ip header again.